assets/js/googlesitekit/widgets/datastore/widgets.js

```diff
@@ -176,88 +177,64 @@ export const actions = {
 
 export const controls = {};
 
-export const reducer = ( state, { type, payload } ) => {
+export const reducer = createReducer( ( draft, { type, payload } ) => {
 	switch ( type ) {
 		case ASSIGN_WIDGET: {
 			const { slug, areaSlugs } = payload;
 
-			const { areaAssignments } = state;
 			areaSlugs.forEach( ( areaSlug ) => {
-				if ( areaAssignments[ areaSlug ] === undefined ) {
-					areaAssignments[ areaSlug ] = [];
+				if ( draft.areaAssignments[ areaSlug ] === undefined ) {
+					draft.areaAssignments[ areaSlug ] = [];
 				}
 
-				if ( ! areaAssignments[ areaSlug ].includes( slug ) ) {
-					areaAssignments[ areaSlug ].push( slug );
+				if ( ! draft.areaAssignments[ areaSlug ].includes( slug ) ) {
+					draft.areaAssignments[ areaSlug ].push( slug );
 				}
 			} );
 
-			return {
-				...state,
-				areaAssignments,
-			};
+			return draft;
 		}
 
 		case REGISTER_WIDGET: {
 			const { slug, settings } = payload;
 
-			if ( state.widgets[ slug ] !== undefined ) {
+			if ( draft.widgets[ slug ] !== undefined ) {
 				global.console.warn(
 					`Could not register widget with slug "${ slug }". Widget "${ slug }" is already registered.`
 				);
 
-				return state;
+				return draft;
 			}
 
-			return {
-				...state,
-				widgets: {
-					...state.widgets,
-					[ slug ]: {
-						...settings,
-						slug,
-					},
-				},
-			};
+			draft.widgets[ slug ] = { ...settings, slug };
+			return draft;
 		}
 
 		case SET_WIDGET_STATE: {
 			const { slug, Component, metadata } = payload;
 
-			return {
-				...state,
-				widgetStates: {
-					...state.widgetStates,
-					[ slug ]: {
-						Component,
-						metadata,
-					},
-				},
-			};
+			draft.widgetStates[ slug ] = { Component, metadata };
+			return draft;
 		}
 
 		case UNSET_WIDGET_STATE: {
 			const { slug, Component, metadata } = payload;
 
-			const widgetStates = { ...state.widgetStates };
 			if (
-				widgetStates?.[ slug ]?.Component === Component &&
-				widgetStates?.[ slug ]?.metadata === metadata
+				draft.widgetStates?.[ slug ]?.Component === Component &&
+				draft.widgetStates?.[ slug ]?.metadata === metadata
 			) {
-				delete widgetStates[ slug ];
```

Comment:
This line should not be removed :)

Reply:
Calling `delete` on a draft works as expected, no need to set it to `undefined`.

Under the bonnet Immer uses Proxy which catches the `delete` - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy/Proxy/deleteProperty

Reply:
In that case, given `delete draft.widgetStates[ slug ];` is used below, does this line really need to stay then?

Reply:
Ah - the comment was correct at the time of writing, but it's been addressed in the recent commit https://github.com/google/site-kit-wp/pull/5936/commits/154a24d273b0f2f10ca9245c33bb7a1ea07b0e27

I'll resolve this convo :)